Bluetooth: Mesh: Add Scene client to dimmer sample (#11164)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/light_dimmer/README.rst

********

This sample can be used to control the state of light sources.
In addition to the on/off functionality, it allows changing the light level (brightness) of an LED light, and storing and recalling scenes of light levels.
To display any functionality, the sample must be paired with a device with the :ref:`mesh light fixture <bluetooth_mesh_light_lc>` sample running in the same mesh network.

The sample instantiates the following models:

 * :ref:`bt_mesh_lvl_cli_readme`
 * :ref:`bt_mesh_onoff_cli_readme`
 * :ref:`bt_mesh_scene_cli_readme`

Devices are nodes with a provisionee role in a mesh network.
Provisioning is performed using the `nRF Mesh mobile app`_.
This mobile application is also used to configure key bindings, and publication and subscription settings of the Bluetooth mesh model instances in the sample.

> After provisioning and configuring the mesh models supported by the sample in the `nRF Mesh mobile app`_, **Button** 1 on the Mesh Light Dimmer device can be used to control the configured network nodes' LEDs, while **Button** 2 can be used to store and restore scenes on the network nodes.

Reviewer: ```suggestion
After provisioning and configuring the mesh models supported by the sample in the `nRF Mesh mobile app`_, **Button 1**  on the Mesh Light Dimmer device can be used to control the configured network nodes' LEDs, while **Button 2**  can be used to store and restore scenes on the network nodes.
```

   On press and hold, **Button 1** will publish a Generic Level move set message using the configured publication parameters of its model instance and will continuously dim the LED lightness state on a :ref:`mesh light fixture <bluetooth_mesh_light_lc>` device. On release, the button publishes another Generic Level move set message with the ``delta`` parameter set to 0 and stops the continuous level change from the previous message.

> Button 1:

Reviewer: ```suggestion
Button 2:
```

>    On short press, **Button 2** will publish a Scene Restore message using the configured publication parameters of its model instance, and restore the LED state of all the targets to the values stored under the current scene number. Each press of the button will recall the next scene, meaning, the first press will recall scene 1, the next press will recall scene 2, and so on, before wrapping around back to scene 0.

Reviewer: Wrap around back to scene 1 maybe? No mention of a scene 0 before

Author: This is indeed wrong, but not for the reason you say. It _does_ wrap to scene 1, but the first press recalls scene 2. This is the way it works because it has to be "on" scene 1 at boot for the store operation on fresh boot to make sense (store to scene 1).

>    On long press, **Button 2** will publish a Scene Store message using the configure publication parameters of its model instance, and store the current LED state of all the targets under the scene with the most recently recalled scene number.

Reviewer: ```suggestion
   On long press, **Button 2** will publish a Scene Store message using the configured publication parameters of its model instance, and store the current LED state of all the targets under the scene with the most recently recalled scene number.
```

>    On long press, **Button 2** will publish a Scene Store message using the configure publication parameters of its model instance, and store the current LED state of all the targets under the scene with the most recently recalled scene number.

Author: Like I mentioned in Stines comment: It sends the message upon release. But "Long press and release" is a good middle ground.

Configure the first element on the target **Mesh Light Fixture** node:

* Bind the following models to **Application Key 1**: Generic Level Server, Generic OnOff Server, Scene Server, and Scene Setup Server.
* Subscribe all models to the same group as set for publication for the **Mesh Light Dimmer**.

You should now be able to perform the following actions:

* Press and release **Button 1** to see the LED on the target Mesh Light Fixture device toggle on and off.
* Press and hold **Button 1** to see the light level of the LED on the target Mesh Light Fixture device slowly decrease or increase.

> * Short press **Button 2** to rotate through scenes, recalling each of them in turn.

Reviewer: ```suggestion
* Press and release **Button 2** to rotate through scenes, recalling each of them in turn.
```

> * Long press **Button 2** to store the current LED states as a scene with the current scene number.

Reviewer: ```suggestion
* Press and hold **Button 2** to store the current LED states as a scene with the current scene number.
```

Reviewer 2: I think we should at least align this the language between the two buttons. Does long press also include release, or is it press and hold. (We should explain short and long press)
```suggestion
* Press and hold **Button 2**, on release the current LED states will be stored with the current scene number.
```
